huaweicloud/resource_huaweicloud_rds_instance_v3.go

```diff
@@ -255,106 +232,259 @@ func resourceRdsInstanceV3() *schema.Resource {
+		VpcId:               d.Get("vpc_id").(string),
+		SubnetId:            d.Get("subnet_id").(string),
+		SecurityGroupId:     d.Get("security_group_id").(string),
+		ConfigurationId:     d.Get("param_group_id").(string),
+		TimeZone:            d.Get("time_zone").(string),
+		FixedIp:             d.Get("fixed_ip").(string),
+		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+		Region:              region,
+	}
 
-	arrayIndex := map[string]int{
-		"backup_strategy": 0,
-		"db":              0,
-		"volume":          0,
+	// var azList []string
+	azList := make([]string, len(d.Get("availability_zone").([]interface{})))
+	for i, az := range d.Get("availability_zone").([]interface{}) {
+		azList[i] = az.(string)
 	}
+	createRdsOpts.AvailabilityZone = strings.Join(azList, ",")
 
-	params, err := buildRdsInstanceV3CreateParameters(opts, arrayIndex)
-	if err != nil {
-		return fmt.Errorf("Error building the request body of api(create), err=%s", err)
+	if hasFilledOpt(d, "db") {
+		db := instances.Datastore{
+			Type:    d.Get("db.0.type").(string),
+			Version: d.Get("db.0.version").(string),
+		}
+		createRdsOpts.Datastore = &db
+		createRdsOpts.Password = d.Get("db.0.password").(string)
+		createRdsOpts.Port = strconv.Itoa(d.Get("db.0.port").(int))
 	}
-	r, err := sendRdsInstanceV3CreateRequest(d, params, client)
-	if err != nil {
-		return fmt.Errorf("Error creating RdsInstanceV3, err=%s", err)
+	if hasFilledOpt(d, "volume") {
+		volume := instances.Volume{
+			Type: d.Get("volume.0.type").(string),
+			Size: d.Get("volume.0.size").(int),
+		}
+		createRdsOpts.Volume = &volume
+		createRdsOpts.DiskEncryptionId = d.Get("volume.0.disk_encryption_id").(string)
+	}
+	if hasFilledOpt(d, "backup_strategy") {
+		backup := instances.BackupStrategy{
+			StartTime: d.Get("backup_strategy.0.start_time").(string),
+			KeepDays:  d.Get("backup_strategy.0.keep_days").(int),
+		}
+		createRdsOpts.BackupStrategy = &backup
+	}
+	if hasFilledOpt(d, "ha_replication_mode") {
+		ha := instances.Ha{
+			Mode:            "ha",
+			ReplicationMode: d.Get("ha_replication_mode").(string),
+		}
+		createRdsOpts.Ha = &ha
 	}
 
-	timeout := d.Timeout(schema.TimeoutCreate)
-	obj, err := asyncWaitRdsInstanceV3Create(d, config, r, client, timeout)
+	res, err := instances.Create(client, createRdsOpts).Extract()
 	if err != nil {
 		return err
 	}
-	id, err := navigateValue(obj, []string{"job", "instance", "id"}, nil)
-	if err != nil {
-		return fmt.Errorf("Error constructing id, err=%s", err)
+	d.SetId(res.Instance.Id)
+
+	log.Printf("[DEBUG] Waiting for instance (%s) to become running", d.Id())
+	// After the rds instance is created, it will enter the backup state.
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD", "ACTIVE"},
+		Target:     []string{"BACKING UP"},
+		Refresh:    InstanceV3StateRefreshFunc(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
```

Comment:
please use `PollInterval` instead.

Reply:
this state check is move from creation function to updation function.
timeout set with 15 second.

Reply:
`PollInterval` is used, set with 15 seconds. Is 20 seconds will be better?
The average creation time is 4 to 5 minutes.